diff --git a/yoga/Yoga.h b/yoga/Yoga.h
index 2a4f205..cce58a3 100644
--- a/yoga/Yoga.h
+++ b/yoga/Yoga.h
@@ -7,6 +7,29 @@
 
 #pragma once
 
+#if !defined(__cpp_lib_bit_cast) || __cpp_lib_bit_cast < 201806L
+#include <cstring>
+#include <type_traits>
+
+namespace std {
+template <class To, class From>
+std::enable_if_t<
+    sizeof(To) == sizeof(From) && std::is_trivially_copyable<From>::value &&
+        std::is_trivially_copyable<To>::value,
+    To>
+bit_cast(const From& src) noexcept {
+  static_assert(
+      std::is_trivially_constructible<To>::value,
+      "This implementation additionally requires "
+      "destination type to be trivially constructible");
+
+  To dst; // object lifetime started
+  std::memcpy(&dst, &src, sizeof(To));
+  return dst;
+}
+} // namespace std
+#endif
+
 /**
  * `#include <yoga/Yoga.h>` includes all of Yoga's public headers.
  */
